### DISCUSSION TITLE
NextJS 13.2.4 Dev Looping and Crashes

### DISCUSSION THREAD

**Discussion by st11x** [vercel/next.js#47424] [Help]
### Summary

Hi,

I upgraded from 13.1 to 13.2.4 yesterday and now Dev mode is very unstable. On certain pages, it keeps compiling and then restarts, you cannot even load the page properly.  This is the log that goes in a loop until you stop going to those pages that can trigger it.

Is there a new setting that I'm missing?

Matt

dev_1     | event - compiled successfully in 624 ms (518 modules)
dev_1     | wait  - compiling...
dev_1     | event - compiled successfully in 1 ms (518 modules)
dev_1     | wait  - compiling...
dev_1     | event - compiled successfully in 2 ms (518 modules)
dev_1     | wait  - compiling...
dev_1     | event - compiled successfully in 1 ms (518 modules)
dev_1     | wait  - compiling...
dev_1     | event - compiled successfully in 1 ms (518 modules)
dev_1     | wait  - compiling...
dev_1     | event - compiled successfully in 1 ms (518 modules)
dev_1     | wait  - compiling...
dev_1     | event - compiled successfully in 3 ms (518 modules)
dev_1     | wait  - compiling...
dev_1     | event - compiled successfully in 1 ms (518 modules)
dev_1     | wait  - compiling...
dev_1     | event - compiled successfully in 2 ms (518 modules)
dev_1     | wait  - compiling...
dev_1     | event - compiled successfully in 2 ms (518 modules)
dev_1     | wait  - compiling...
dev_1     | event - compiled successfully in 1 ms (518 modules)
dev_1     | wait  - compiling...
dev_1     | event - compiled successfully in 2 ms (518 modules)
dev_1     | warn  - The server is running out of memory, restarting to free up memory.
dev_1     | ready - started server on 0.0.0.0:3000, url: http://localhost:3000
dev_1     | warn  - You have enabled experimental feature (appDir) in next.config.js.

**Comment by icyJoseph** [comment]
This is most likely something to do with your environment. What NodeJS version do you have?

Have you tried to reinstall all of your modules again? Maybe even delete the lock file?

**Reply by st11x** [reply]
I'm on Node v16.15.0 and it is a docker container from the image node:lts-alpine. React is at 18.2. 

It seems like it is more likely to do this when there is an error on the page, however small that is. For example, I was using a react icon (CiHospital1)  that was flagged as using a dataName attribute that is incorrect. That also contributed to this.

**Reply by st11x** [reply]
I just upgraded to Node v18.15.0 but I'm still facing the same issue.

**Reply by st11x** [reply]
I was distilling my app down to the bare minimum so I can create an example and now on this simple page, it worked for a while and then all of a sudden, the page started refreshing and thus caused these log entries; the server eventually crashed. As this is a very simple page, it took a longer time for it to exhaust the server memory.

I saw this in my network tab, it is a rapid fire of requests to the page without any activity/interaction on it. 

![image](https://user-images.githubusercontent.com/7398499/227724994-bc093413-a79a-4e08-a5a4-ffe4085d5852.png)

All these as I was debugging the placement of this searchbox

![image](https://user-images.githubusercontent.com/7398499/227725196-4ae49ef1-a394-410b-873c-cf0cc41b7080.png)

**Comment by DmitriyPast** [comment]
Happened to me on Next.js 16.0.1...
I HATE React!

**Reply by icyJoseph** [reply]
Hi, do you have a custom distDir in your next config file?
